cmake/alpakaCommon.cmake

```diff
@@ -407,6 +407,15 @@ if(ALPAKA_ACC_GPU_CUDA_ENABLE)
         elseif(CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA")
             message(STATUS "nvcc is used as CUDA compiler")
 
+            # If host compiler is clang < 8, nvcc does not forward -std=c++17
+            if (CMAKE_CUDA_HOST_COMPILER MATCHES ".*clang\\+\\+.*")
+                execute_process(COMMAND ${CMAKE_CUDA_HOST_COMPILER} --version OUTPUT_VARIABLE CLANG_VERSION_STRING)
+                string(REGEX MATCH "clang version ([0-9]+)\\.[0-9]+\\.[0-9]+" DUMMY ${CLANG_VERSION_STRING})
+                if (CMAKE_MATCH_1 LESS 8)
+                    alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-std=c++17>)
+                endif ()
+            endif ()
+
```

Comment:
Unfortunately, CMake does not detect and provide better variables to inspect the CUDA host compiler.